### PR TITLE
[Tool] Enable sasl option with librdkafka

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [x] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by kevincai — DISMISSED** [review]

**Review by imay — DISMISSED** [review]

File: thirdparty/build-thirdparty.sh [hunk 1]
```diff
@@ -514,14 +514,29 @@ build_rocksdb() {
...
+
+    cd $TP_SOURCE_DIR/$SASL_SOURCE
+    sh ./autogen.sh
+    ./configure --prefix=$TP_INSTALL_DIR  --enable-static=yes
+    make -j$PARALLEL
+    make install
+    cd $TP_INSTALL_DIR/lib
+    ls | grep sasl | grep so | xargs rm -rf
+    cd $TP_INSTALL_DIR/lib/sasl2
+    find . | grep so | xargs rm -rf
```
Comment (kevincai) [inline thirdparty/build-thirdparty.sh:529]:
```suggestion
    ./autogen.sh --prefix=$TP_INSTALL_DIR  --enable-static=yes --enable-shared=no
    make -j$PARALLEL
    make install
```

Comment (kevincai) [reply thirdparty/build-thirdparty.sh:529]: also need to point the openssl install location and gssapi location, or disable it explicitly.

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — DISMISSED** [review]

File: thirdparty/build-thirdparty.sh [hunk 2]
```diff
@@ -514,14 +514,26 @@ build_rocksdb() {
     export CFLAGS=$OLD_FLAGS
 }
 
+# sasl
+build_sasl() {
+    check_if_source_exist $SASL_SOURCE
+    cd $TP_SOURCE_DIR/$SASL_SOURCE
+    OLD_FLAGS=$CFLAGS
+    unset CFLAGS
+    ./autogen.sh --prefix=$TP_INSTALL_DIR --enable-gssapi=no --enable-static=yes --enable-shared=no
```
Comment (kevincai) [inline thirdparty/build-thirdparty.sh:523]:
```suggestion
    CFLAGS= ./autogen.sh --prefix=$TP_INSTALL_DIR --enable-gssapi=no --enable-static=yes --enable-shared=no
```

can use `CFLAGS= ...` to force reset the env effectively just the specific line.

File: thirdparty/build-thirdparty.sh [hunk 3]
```diff
@@ -514,14 +514,26 @@ build_rocksdb() {
...
+
 # librdkafka
 build_librdkafka() {
     check_if_source_exist $LIBRDKAFKA_SOURCE
 
     cd $TP_SOURCE_DIR/$LIBRDKAFKA_SOURCE
 
     $CMAKE_CMD -DCMAKE_LIBRARY_PATH=$TP_INSTALL_DIR/lib -DCMAKE_INCLUDE_PATH=$TP_INSTALL_DIR/include \
-        -DBUILD_SHARED_LIBS=0 -DCMAKE_INSTALL_PREFIX=$TP_INSTALL_DIR -DRDKAFKA_BUILD_STATIC=ON -DWITH_SASL=OFF -DWITH_SASL_SCRAM=ON \
+        -DBUILD_SHARED_LIBS=0 -DCMAKE_INSTALL_PREFIX=$TP_INSTALL_DIR -DRDKAFKA_BUILD_STATIC=ON -DWITH_SASL=ON -DWITH_SASL_SCRAM=ON \
```
Comment (chaoyli) [inline thirdparty/build-thirdparty.sh:536]: What's the usage of SASL for librdkafka?

Comment (zaorangyang) [reply thirdparty/build-thirdparty.sh:536]: > What's the usage of SASL for librdkafka?

To support more authentication methods

File: thirdparty/build-thirdparty.sh (hunk 2 above)
Comment (zaorangyang) [reply thirdparty/build-thirdparty.sh]: > can use `CFLAGS= ...` to force reset the env effectively just the specific line.

updated

**Review by kevincai — DISMISSED** [review]

**Review by meegoo — DISMISSED** [review]

File: thirdparty/build-thirdparty.sh [hunk 4]
```diff
@@ -514,14 +514,23 @@ build_rocksdb() {
     export CFLAGS=$OLD_FLAGS
 }
 
+# sasl
+build_sasl() {
+    check_if_source_exist $SASL_SOURCE
+    cd $TP_SOURCE_DIR/$SASL_SOURCE
+    CFLAGS=  ./autogen.sh --prefix=$TP_INSTALL_DIR --enable-gssapi=no --enable-static=yes --enable-shared=no
```
Comment (chaoyli) [inline thirdparty/build-thirdparty.sh:521]: the format is not right. 
```suggestion
    CFLAGS=./autogen.sh --prefix=$TP_INSTALL_DIR --enable-gssapi=no --enable-static=yes --enable-shared=no
```

**Review by chaoyli — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Pull request merged**